go/kbhttp/srv.go

```diff
@@ -128,26 +127,20 @@ func (h *Srv) Start() (err error) {
 		Addr:    address,
 		Handler: h.ServeMux,
 	}
-	go func() {
-		h.Lock()
-		h.active = true
-		h.Unlock()
+	go func(server *http.Server) {
```

Comment:
It's possible that one could call Start, then Stop, but that the goroutine spawned by Start would run after the Stop. I guess that was already possible, though.

Reply:
Yeah this PR is mostly about making that case better. If that happens, `Serve` should return an error and that goroutine should just go away. Any `Serve` call after `Close` fails with `ErrServerClosed`